source/core/parameters_lagrangian.cc

```diff
@@ -1772,6 +1772,128 @@ namespace Parameters
       prm.leave_subsection();
     }
 
+
+    template <int dim>
+    void
+    ParticleRayTracing<dim>::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("particle ray tracing");
+      {
+        prm.declare_entry("starting photon insertion position",
+                          "0.,0.,0.",
+                          Patterns::List(Patterns::Double()),
+                          "Location of the first photon being inserted.");
+
+        prm.declare_entry("first insertion direction",
+                          "1.,0.,0.",
+                          Patterns::List(Patterns::Double()),
+                          "First direction used to insert photons.");
+        prm.declare_entry("second insertion direction",
+                          "0.,1.,0.",
+                          Patterns::List(Patterns::Double()),
+                          "photon displacement direction .");
+        prm.declare_entry("photon displacement vector",
+                          "0.,0.,1.",
+                          Patterns::List(Patterns::Double()),
+                          "Displacement vector of each photons.");
+
+        prm.declare_entry("number of photon first direction",
+                          "1",
+                          Patterns::Integer(),
+                          "Number of photons that will be inserted "
+                          "in the first direction.");
+        prm.declare_entry("number of photon second direction",
+                          "1",
+                          Patterns::Integer(),
+                          "Number of photons that will be inserted "
+                          "in the second direction.");
+
+        prm.declare_entry("distance between photon first direction",
+                          "1.",
+                          Patterns::Double(),
+                          "Distance between consecutive photon in "
+                          "the first direction.");
+
+        prm.declare_entry("distance between photon second direction",
+                          "1.",
+                          Patterns::Double(),
+                          "Distance between consecutive photon in "
+                          "the second direction.");
+
```

Comment:
Will clearly be better as lists!

Reply:
yeah, just a list. We have all the tools to parse lists. In the end when we want to enumerate stuff, it would be much bettter if we would pass them as lists